x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -1233,24 +1256,109 @@ public TypedData get() {
 
     /**
      * Holds a data value and the intended parse type of that value
-     * @param data - value to test against
-     * @param type - type of the value, for building expressions
-     * @param name - a name for the value, used for generating test case names
      */
-    public record TypedData(Object data, DataType type, String name) {
-
+    public static class TypedData {
```

Comment:
I had to un-record this because I wanted the ctor with all the stuff in it to be private.